includes/Story_Post_Type.php

```diff
@@ -753,18 +754,19 @@ public function redirect_post_type_archive_urls( $bypass, $query ) {
 
 		// 'pagename' is for most permalink types, name is for when the %postname% is used as a top-level field.
 		if ( 'stories' === $query->get( 'pagename' ) || 'stories' === $query->get( 'name' ) ) {
-			if ( $query->get( 'feed' ) ) {
-				$feed                  = ( $query->get( 'feed' ) === 'feed ' ) ? $query->get( 'feed' ) : '';
-				$post_type_archive_url = get_post_type_archive_feed_link( self::POST_TYPE_SLUG, $feed );
-			} else {
-				$post_type_archive_url = get_post_type_archive_link( self::POST_TYPE_SLUG );
+			$redirect_url = get_post_type_archive_link( self::POST_TYPE_SLUG );
+			if ( $query->get( 'page' ) && self::POST_TYPE_SLUG === get_post_type( absint( $query->get( 'page' ) ) ) ) {
```

Comment:
```suggestion
			if ( $query->get( 'page' ) && is_numeric( $query->get( 'page' ) ) && self::POST_TYPE_SLUG === get_post_type( absint( $query->get( 'page' ) ) ) ) {
```

Reply:
This isn't just a nice to have, it could save a database query, as `get_post_type` would not be in the cache at this point. A query on every 404 page, could result in a lot of database traffic.

Reply:
`is_numeric` or `is_int`?

Reply:
I think it has be `is_numeric` is that this will be a string as this point. But it mean that `12.0` be cast to `12`.

Reply:
Check added